core/src/main/java/io/lionweb/lioncore/java/serialization/DeserializationStatus.java

```diff
@@ -35,26 +40,65 @@ class DeserializationStatus {
 
   DeserializationStatus(
       List<SerializedClassifierInstance> originalList,
-      ClassifierInstanceResolver outsideInstancesResolver) {
+      ClassifierInstanceResolver outsideInstancesResolver,
+      PrimitiveValuesSerialization primitiveValuesSerialization) {
+    this.primitiveValuesSerialization = primitiveValuesSerialization;
     sortedList = new ArrayList<>();
     nodesToSort = new ArrayList<>(originalList);
     this.proxiesInstanceResolver = new LocalClassifierInstanceResolver();
     this.globalInstanceResolver =
         new CompositeClassifierInstanceResolver(outsideInstancesResolver, proxiesInstanceResolver);
   }
 
+  public Property getProperty(Classifier<?> classifier, MetaPointer metaPointer) {
+    featuresCache.computeIfAbsent(classifier, c -> new HashMap<>());
+    Map<MetaPointer, Feature<?>> featuresMap = featuresCache.get(classifier);
+    featuresMap.computeIfAbsent(metaPointer, classifier::getPropertyByMetaPointer);
```

Comment:
I think we can even compress this further -- `return featuresMap.computeIfAbsent()` should work.

Reply:
Absolutely. We can also compress the previous two lines in one (and similarly for the other two methods)